test/bdd/steps/queries.py

```diff
@@ -530,3 +530,9 @@ def check_for_duplicates(context, neg):
         assert not has_dupe, "Found duplicate for %s" % (dup, )
     else:
         assert has_dupe, "No duplicates found"
+
+@then(u'the result page will include (?P<error>.*).')
+def error_check(context,error):
+    if not error in getattr(context.response,'page'):
+        assert not error, "Incorrect error displayed. Expected page to contain \"%s\" but got %s" % (error, vars(context.response))
+
```

Comment:
Just for the syntax, it should be something like that:

def error_check(..):
    assert_in(error, context.response.page, "some error")

Have a look at https://docs.python.org/3/library/unittest.html#assert-methods for the different asserts. The underscore notation comes from the nose library.

On a more general note, you should rather extend the existing 'a HTTP xxx is returned' to 'a HTTP xxx is returned with error "<msg>"'. (Just saying in case you hadn't planed that anyways.)